source/common/router/config_impl.h

```diff
@@ -324,7 +324,7 @@ class VirtualHostImpl : public VirtualHost, Logger::Loggable<Logger::Id::router>
   RouteConstSharedPtr
   getRouteFromRoutes(const RouteCallback& cb, const Http::RequestHeaderMap& headers,
                      const StreamInfo::StreamInfo& stream_info, uint64_t random_value,
-                     std::vector<RouteEntryImplBaseConstSharedPtr> routes) const;
+                     absl::Span<const RouteEntryImplBaseConstSharedPtr> routes) const;
```

Comment:
is this `const` necessary?

Reply:
Yes. Otherwise, the vector can't be converted to absl::Span:
```
source/common/router/config_impl.cc:1920:75: error: no matching constructor for initialization of 'absl::Span<RouteEntryImplBaseConstSharedPtr>' (aka 'Span<shared_ptr<const Envoy::Router::RouteEntryImplBase>>')

source/common/router/config_impl.cc:1925:75: error: no viable conversion from 'const std::vector<RouteEntryImplBaseConstSharedPtr>' (aka 'const vector<shared_ptr<const Envoy::Router::RouteEntryImplBase>>') to 'absl::Span<RouteEntryImplBaseConstSharedPtr>' (aka 'Span<shared_ptr<const Envoy::Router::RouteEntryImplBase>>')

source/common/router/config_impl.cc:1936:69: error: no viable conversion from 'const std::vector<RouteEntryImplBaseConstSharedPtr>' (aka 'const vector<shared_ptr<const Envoy::Router::RouteEntryImplBase>>') to 'absl::Span<RouteEntryImplBaseConstSharedPtr>' (aka 'Span<shared_ptr<const Envoy::Router::RouteEntryImplBase>>')
```

Reply:
Hmmm, a little weird. The `Span` should be const by default just like the `absl::string_view`. Anyway, it isn't big matter.